src/ScrollingContainer.js

```diff
@@ -270,15 +266,44 @@ ScrollingContainer.prototype.initScrolling = function () {
         container.position[direction] = Math.round(Position[direction]);
 
         self.updateScrollBars();
-
     };
 
-    this._checkLimit = function (limitName) {
-        if (!(limitName in limitsReached)) {
-            limitsReached[limitName] = true;
-            self.emit(limitName + "Overflow");
+    this._checkLimits = function (direction) {
+        if (!this.scrolling && Math.round(Speed[direction]) === 0) {
+            return;
         }
-    };
+
+        var bounds = this.getInnerBounds();
+
+        var minimumLimit = (direction == "y") ?
+            Math.round(Math.min(0, this._height - bounds.height)) :
+            Math.round(Math.min(0, this._width - bounds.width));
+
+        var maxLimit = targetPosition[direction] >= 0;
+        var minLimit = targetPosition[direction] <= minimumLimit;
+
+        if (direction === "x" && maxLimit) {
+            self.emit("leftReached");
+        } else if (direction === "x" && minLimit) {
+            self.emit("rightReached");
+        } else if (direction === "y" && maxLimit) {
+            self.emit("topReached");
+        } else if (direction === "y" && minLimit) {
+            self.emit("bottomReached");
+        }
+
+        if ((maxLimit || minLimit) && self._boundsChanged(bounds)) {
+            self._checkLimits(direction);
```

Comment:
If the user is reacting to the events sent and it is adding new content to the scrolling area.
This ensures 2 things:

The bounds are forced to be updated
It keeps rechecking the limits until the bounds do not change this is useful to prevent a little bouncing back effect if the user tries to scroll too fast.